l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -194,6 +195,10 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
 
     /// @dev Used to set the assedAddress for a given assetInfo.
     function setAssetAddress(bytes32 _additionalData, address _assetAddress) external {
+        require(
+            (msg.sender == address(nativeTokenVault)) || (msg.sender == owner()),
```

Comment:
Why should we restrict this?

Reply:
If anyone can override this mapping, this could be used as an attack vector, but we could be less strict, allow more use cases in the future

Reply:
the assetInfo  contains the msg.sender is the hash, so it is not possible to override,

Reply:
I agree, but when bridging through non-authorized vaults, users might be tricked to send data anywhere (phishing attack using the mirror of legit bridge) as he'll still interact with bridgehub address, which seems legit.

Reply:
But then we cannot allow custom bridges

Reply:
updated